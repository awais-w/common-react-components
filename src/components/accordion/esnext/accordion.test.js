import React from 'react';
import { shallow } from 'enzyme';
import uuid from 'uuid';
import Accordion from './accordion';

const defaultProps = {
  allowMultipleOpen: false,
  moreAvailable: true,
  accordionData: [
    {
      header: {
        icon: 'forum',
        title1: '13 Jan 2020, 00:32:21',
        title2: 'Order Number: 6000002600',
        actions: [
          { icon: 'icon-url', isCustom: true, color: 'blue', tooltip: 'copy url' },
          { icon: 'zoom_out_map', color: 'blue', tooltip: 'expand transcript' },
        ],
      },
      body: 'This is card content.',
    },
  ],
};

// Mock the uuid
uuid.v4 = jest.fn(() => '00000000-0000-0000-0000-000000000000');

const setup = ({ props }) => {
  const component = shallow(<Accordion {...props} />);

  return {
    wrapper: component,
  };
};

describe('<AccordionCard />', () => {
  describe('with default props', () => {
    const component = setup({ props: defaultProps });

    it('should match default snapshot', () => {
      expect(component.wrapper).toMatchSnapshot();
    });

    it('exists', () => expect(component.wrapper).toExist());
  });
});
