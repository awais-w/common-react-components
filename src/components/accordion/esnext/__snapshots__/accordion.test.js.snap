// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccordionCard /> with default props should match default snapshot 1`] = `
<div
  className="ac-accordion"
>
  <AccordionCard
    allowMultipleOpen={false}
    cardData={
      Object {
        "body": "This is card content.",
        "header": Object {
          "actions": Array [
            Object {
              "color": "blue",
              "icon": "icon-url",
              "isCustom": true,
              "tooltip": "copy url",
            },
            Object {
              "color": "blue",
              "icon": "zoom_out_map",
              "tooltip": "expand transcript",
            },
          ],
          "icon": "forum",
          "title1": "13 Jan 2020, 00:32:21",
          "title2": "Order Number: 6000002600",
        },
      }
    }
    isExpanded={false}
    key="00000000-0000-0000-0000-000000000000"
    onCardClick={[Function]}
  />
  <Button
    buttonWidth="containerFit"
    icon="playlist_add"
    iconMode="left"
    isDisabled={false}
    isLoading={false}
    isSecondary={true}
    isShort={true}
    loadingText=""
    onClick={[Function]}
    text="Load More"
    type="button"
  />
</div>
`;
